autogluon/task/image_classification/classifier.py

```diff
@@ -136,15 +137,15 @@ def predict_img(img, ensemble=False):
                 return ind, probai, proba
 
         def avg_prediction(different_dataset, threshold=0.001):
-            result = {}
+            result = defaultdict(list)
             inds, probas, probals_all = [], [], []
             for i in range(len(different_dataset)):
-                items = len(different_dataset[0])
-                for j in range(items):
-                    result.setdefault(j, []).append(different_dataset[i][j])
+                for j in range(len(different_dataset[0])):
+                    result[j].append(different_dataset[i][j])
+
             for c in result.keys():
                 proba_all = sum([*result[c]]) / len(different_dataset)
-                proba_all = mx.nd.array(np.where(proba_all.asnumpy() >= threshold, proba_all.asnumpy(), 0))
+                proba_all = (proba_all > threshold) * proba_all
```

Comment:
@aptsunny This is not equivalent functionality to my understanding. Why did you change it?

Reply:
Actually, **proba_all** can all be filtered by threshold, but the modification is faster than before，it should be the equivalent function implement.

Reply:
Just a small comment, your replaced `>=` with `>`, is that fine ? The rest behavior will be the same, since all the rest will be `False` which is equivalent to `0`,  so multiplying with proba_all will not change 0s.

Reply:
Thank you very much, I ignored this detail.

Reply:
Thanks for fixing!